.gitlab/e2e.yml

```diff
@@ -21,59 +21,61 @@
   - export DATADOG_AGENT_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_app_key --with-decryption --query "Parameter.Value" --out text)
   - export DATADOG_AGENT_RC_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_rc_key --with-decryption --query "Parameter.Value" --out text)
 
-k8s-e2e-cws-dev:
+.k8s_e2e_template_dev:
   extends: .k8s_e2e_template
-  rules: !reference [.on_dev_branch_manual]
-  needs: []
+  needs:
+    - dev_branch_multiarch-a7
+    - dca_dev_branch
   script:
     - !reference [.k8s-e2e-cws-cspm-init]
-    - inv -e e2e-tests --agent-image=datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3 --dca-image=datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG} --argo-workflow=cws
+    - inv -e e2e-tests --agent-image=datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3 --dca-image=datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG} --argo-workflow=$ARGO_WORKFLOW
 
-k8s-e2e-cws-main:
+.k8s_e2e_template_main:
   extends: .k8s_e2e_template
-  rules: !reference [.on_main]
-  # needs:
-  #   - dev_master-a6
-  #   - dev_master-a7
-  retry: 1
+  needs:
+    - dev_master-a7
+    - dca_dev_master
   script:
     - !reference [.k8s-e2e-cws-cspm-init]
-    - inv -e e2e-tests --agent-image=datadog/agent-dev:master-py3 --dca-image=datadog/cluster-agent-dev:master --argo-workflow=cws
+    - inv -e e2e-tests --agent-image=datadog/agent-dev:master-py3 --dca-image=datadog/cluster-agent-dev:master --argo-workflow=$ARGO_WORKFLOW
+
+k8s-e2e-cws-dev:
+  extends: .k8s_e2e_template_dev
+  rules: !reference [.on_dev_branch_manual]
+  variables:
+    ARGO_WORKFLOW: cws
+
+k8s-e2e-cws-main:
+  extends: .k8s_e2e_template_main
+  rules: !reference [.on_main]
+  retry: 1
+  variables:
+    ARGO_WORKFLOW: cws
 
 k8s-e2e-cspm-dev:
-  extends: .k8s_e2e_template
+  extends: .k8s_e2e_template_dev
   rules: !reference [.on_dev_branch_manual]
-  needs: []
-  script:
-    - !reference [.k8s-e2e-cws-cspm-init]
-    - inv -e e2e-tests --agent-image=datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3 --dca-image=datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG} --argo-workflow=cspm
+  variables:
+    ARGO_WORKFLOW: cspm
 
 k8s-e2e-cspm-main:
-  extends: .k8s_e2e_template
+  extends: .k8s_e2e_template_main
   rules: !reference [.on_main]
-  # needs:
-  #   - dev_master-a6
-  #   - dev_master-a7
   retry: 1
-  script:
-    - !reference [.k8s-e2e-cws-cspm-init]
-    - inv -e e2e-tests --agent-image=datadog/agent-dev:master-py3 --dca-image=datadog/cluster-agent-dev:master --argo-workflow=cspm
+  variables:
+    ARGO_WORKFLOW: cspm
 
 k8s-e2e-otlp-dev:
-  extends: .k8s_e2e_template
+  extends: .k8s_e2e_template_dev
   rules: !reference [.on_dev_branch_manual]
-  needs: [] # we can't explicitly define the dependencies because a job cannot depend on other manual jobs.
```

Comment:
According to [this article](https://about.gitlab.com/blog/2021/05/20/dag-manual-fix/) this should be possible now